apps/remix-ide/src/app/panels/file-panel.js

```diff
@@ -102,11 +104,20 @@ module.exports = class Filepanel extends ViewPlugin {
     if (!item) throw new Error('Invalid register context menu argument')
     if (!item.name || !item.id) throw new Error('Item name and id is mandatory')
     if (!item.type && !item.path && !item.extension && !item.pattern) throw new Error('Invalid file matching criteria provided')
-
+    if (this.registeredMenuItems.filter((o) => {
+      return o.id === item.id & o.name === item.name
+    }).length) throw new Error(`Action ${item.name} already exists on ${item.id}`)
     this.registeredMenuItems = [...this.registeredMenuItems, item]
     this.renderComponent()
   }
 
+  removePluginActions (plugin) {
+    this.registeredMenuItems = this.registeredMenuItems.filter((item) => {
+      return item.id !== plugin.name || item.sticky === true
```

Comment:
What do you mean by `item.sticky`?

Reply:
It remains in the context menu, so that when clicked the plugin will be activated and the method called. this allows us to for example include methods at app startup or any other phase. For example for dgit that would be useful. when the plugin is deactivated it doesn't dissappear.

Reply:
Ah! Alright.